src/protocol/http/org/apache/jmeter/protocol/http/control/HttpMirrorServer.java

```diff
@@ -159,14 +192,84 @@ public Exception getException(){
+            return;
+        }
+
+        if (parser.getArgumentById(OPTIONS_OPT) != null) {
+            System.out.println(CLUtil.describeOptions(options).toString());//NOSONAR
+            return;
+        }
+
         int port = HttpMirrorControl.DEFAULT_PORT;
-        if (args.length > 0){
-            port = Integer.parseInt(args[0]);
+
+        if (parser.getArgumentById(PORT_OPT) != null) {
+            CLOption option = parser.getArgumentById(PORT_OPT);
+            String value = option.getArgument(0);
+            try {
+                port = Integer.parseInt(value);
+            } catch (NumberFormatException ignored) {
+            }
+        } else if (args.length > 0) {
+            try {
+                port = Integer.parseInt(args[0]);
+            } catch (NumberFormatException ignored) {
+            }
+        }
+
+        if (System.getProperty("log4j.configurationFile") == null) {// $NON-NLS-1$
+            Configurator.setRootLevel(Level.INFO);
         }
-        LoggingManager.setPriority("INFO"); // default level
-        LoggingManager.setLoggingLevels(System.getProperties() ); // allow override by system properties
+
+        List<CLOption> clOptions = parser.getArguments();
+
+        for (CLOption option : clOptions) {
+            String name = option.getArgument(0);
+            String value = option.getArgument(1);
+
+            switch (option.getDescriptor().getId()) {
+            case LOGLEVEL_OPT:
+                if (!value.isEmpty()) { // Set category
+                    final Level logLevel = Level.getLevel(value);
+                    if (logLevel != null) {
+                        String loggerName = name;
+                        if (name.startsWith("jmeter") || name.startsWith("jorphan")) {
+                            loggerName = "org.apache." + name;// $NON-NLS-1$
+                        }
+                        getLogger().info("Setting log level to " + value + " for " + loggerName);// $NON-NLS-1$ // $NON-NLS-2$
+                        Configurator.setAllLevels(loggerName, logLevel);
+                    } else {
+                        getLogger().warn("Invalid log level, '" + value + "' for '" + name + "'.");// $NON-NLS-1$ // $NON-NLS-2$
+                    }
+                } else { // Set root level
+                    final Level logLevel = Level.getLevel(name);
+                    if (logLevel != null) {
+                        getLogger().info("Setting root log level to " + name);// $NON-NLS-1$
+                        Configurator.setRootLevel(logLevel);
+                    } else {
+                        getLogger().warn("Invalid log level, '" + name + "' for the root logger.");// $NON-NLS-1$ // $NON-NLS-2$
+                    }
+                }
+                break;
+            }
+        }
+
         HttpMirrorServer serv = new HttpMirrorServer(port);
         serv.start();
     }
+
+    private static Logger getLogger() {
```

Comment:
It's starting a server accepting a socket, on which it handles in multi-threaded worker env. Anyway, the logger has been a static member shared by all. Shouldn't be a problem.